src/Text/Pandoc/Readers/Org/DocumentTree.hs

```diff
@@ -161,17 +166,40 @@ headlineToBlocks hdln = do
   let tags = headlineTags hdln
   let text = headlineText hdln
   let level = headlineLevel hdln
-  shouldNotExport <- hasDoNotExportTag tags
   case () of
-    _ | shouldNotExport -> return mempty
     _ | any isArchiveTag  tags -> archivedHeadlineToBlocks hdln
     _ | isCommentTitle text    -> return mempty
     _ | maxLevel <= level      -> headlineToHeaderWithList hdln
     _ | otherwise              -> headlineToHeaderWithContents hdln
 
-hasDoNotExportTag :: Monad m => [Tag] -> OrgParser m Bool
-hasDoNotExportTag tags = containsExcludedTag . orgStateExcludedTags <$> getState
-  where containsExcludedTag s = any (`Set.member` s) tags
+removeExplicitlyExcludedNodes :: Headline -> OrgParserState -> Headline
+removeExplicitlyExcludedNodes hdln st =
+  hdln { headlineChildren =
+           [removeExplicitlyExcludedNodes childHdln st |
+              childHdln <- headlineChildren hdln,
+              not $ any (`Set.member` orgStateExcludedTags st)
+                        (headlineTags childHdln)] }
+
+includeRootAndSelected :: Headline -> OrgParserState -> Headline
+includeRootAndSelected hdln st =
+  hdln { headlineChildren = mapMaybe (`includeAncestorsAndSelected` st)
+                                     (headlineChildren hdln)}
+
+checkForSelectedTags :: Headline -> OrgParserState -> Bool
```

Comment:
Not sure about this name, it doesn't seem clear to me if the result is True or False if a select tag is found. What do you think about `containsSelectTags`?